src/main/java/kristar/projects/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,70 @@
+package kristar.projects.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.nio.file.AccessDeniedException;
+import kristar.projects.dto.shoppingcartdto.AddCartItemRequestDto;
+import kristar.projects.dto.shoppingcartdto.ShoppingCartResponseDto;
+import kristar.projects.dto.shoppingcartdto.UpdateCartItemRequestDto;
+import kristar.projects.services.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "ShoppingCart management", description = "Endpoints for managing shoppingCarts")
+@RestController
+@RequestMapping("api/cart")
```

Comment:
add `server.servlet.context-path=/api` to application.properties
```suggestion
@RequestMapping("/cart")
```